### PR TITLE
Redis connection retry on failure

### PR DESCRIPTION
**Pull request by vdatcu**
- Upon failing to connect to Redis, the `RedisConnectionProvider` script will attempt to retrieve a new upstream host and port and retry the connection
- Added docs for RedisConnectionProvider

### CODE REVIEW

File: src/lua/api-gateway/redis/redisConnectionProvider.lua [hunk 1]
```diff
@@ -76,10 +81,16 @@ function RedisConnectionProvider:getConnection(connection_options)
     end
 
     local redisHost, redisPort = self:getRedisUpstream(redisUpstream)
-    return self:connectToRedis(redisHost, redisPort, redisPassword, redisTimeout)
+    local status, redisInstance = self:connectToRedis(redisHost, redisPort, redisPassword, redisTimeout)
```
Comment (atrifan) [inline src/lua/api-gateway/redis/redisConnectionProvider.lua]: is this only for connection? no retry on redis methods?

**Review by atrifan — CHANGES_REQUESTED** [review]

**Review by andralungu — APPROVED** [review]
👍

**Comment by vdatcu** [comment]
@atrifan @andralungu @corinapurcarea please check

File: src/lua/api-gateway/redis/redisConnectionProvider.lua [hunk 2]
```diff
@@ -76,10 +81,18 @@ function RedisConnectionProvider:getConnection(connection_options)
     end
 
     local redisHost, redisPort = self:getRedisUpstream(redisUpstream)
-    return self:connectToRedis(redisHost, redisPort, redisPassword, redisTimeout)
+    ngx.log(ngx.WARN, "Trying with: " .. tostring(redisHost) .. " and " .. tostring(redisPort))
```
Comment (constantincristian) [inline src/lua/api-gateway/redis/redisConnectionProvider.lua]: First "try" should be logged as `DEBUG`?

Comment (vdatcu) [reply src/lua/api-gateway/redis/redisConnectionProvider.lua]: Yes, definitely. I will update now.

File: src/lua/api-gateway/redis/redisConnectionProvider.lua [hunk 3]
```diff
@@ -76,10 +81,18 @@ function RedisConnectionProvider:getConnection(connection_options)
...
+    local status, redisInstance = self:connectToRedis(redisHost, redisPort, redisPassword, redisTimeout)
+    if not status then
+        -- retry
+        ngx.log(ngx.WARN, "Connection to Redis failed. Retrieving new Redis host and retrying")
+        redisHost, redisPort = self:getRedisUpstream(redisUpstream)
+        ngx.log(ngx.WARN, "Got new upstream: " .. tostring(redisHost) .. " and " .. tostring(redisPort))
```
Comment (constantincristian) [inline src/lua/api-gateway/redis/redisConnectionProvider.lua]: Should be logged as `DEBUG`?

Comment (vdatcu) [reply src/lua/api-gateway/redis/redisConnectionProvider.lua]: Done.

**Pull request merged**